detect/gcp/go.mod

```diff
@@ -0,0 +1,8 @@
+module go.opentelemetry.io/contrib/detect/gcp
```

Comment:
Minor ask, but wondering if we can rename the `detect` directory to `detectors`?